client/packages/system/src/Item/DetailView/Tabs/PackVariants.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { FC } from 'react';
+import { VariantFragment } from '../../api';
+import {
+  Box,
+  DataTable,
+  TableProvider,
+  useColumns,
+  createTableStore,
+  NothingHere,
+  useTranslation,
+  ButtonWithIcon,
+  PlusCircleIcon,
+  Grid,
+  useEditModal,
+  useAuthContext,
+  useNotification,
+  UserPermission,
+} from '@openmsupply-client/common';
+import { usePackVariant } from '../../context';
+import { PackVariantEditModal } from '../../Components/PackVariantEditModal';
+
+const PackVariantTable: FC<{ itemId?: string }> = ({ itemId }) => {
+  const t = useTranslation('catalogue');
+  const { variantsControl } = usePackVariant(itemId ?? '', null);
+  const { isOpen, entity, mode, onClose, onOpen } =
+    useEditModal<VariantFragment>();
+  const { info } = useNotification();
+  const { userHasPermission } = useAuthContext();
+  const infoSnack = info(t('auth.permission-denied'));
+  const hasPermission = userHasPermission(
+    UserPermission.ItemNamesCodesAndUnitsMutate
+  );
+
+  const columns = useColumns<VariantFragment>([
+    'packSize',
+    {
+      key: 'abbreviation',
+      label: 'label.abbreviation',
+      accessor: ({ rowData }) => rowData?.shortName,
+      sortable: false,
+    },
+    {
+      key: 'longName',
+      label: 'label.name',
+      accessor: ({ rowData }) => rowData?.longName,
+      sortable: false,
+    },
+  ]);
+
+  return (
+    <>
+      {isOpen && (
+        <PackVariantEditModal
+          mode={mode}
+          isOpen={isOpen}
+          onClose={onClose}
+          packVariant={entity}
+          itemId={itemId ?? ''}
+        />
+      )}
+      <Box display="flex" justifyContent="flex-end" paddingBottom={2}>
+        <ButtonWithIcon
+          Icon={<PlusCircleIcon />}
+          label={t('label.new-pack-variant')}
+          onClick={hasPermission ? () => onOpen() : () => infoSnack()}
```

Comment:
Made the no permissions message an infoSnack since errorSnack seems too angry for this...

Reply:
ah. I had an earlier comment which perhaps you should ignore. I thought info was not angry enough.
warning?